src/DoubleTheDonation/AddonServiceProvider.php

```diff
@@ -65,10 +65,10 @@ private function loadBackend() {
 		// Register settings page
 		SettingsPage::registerPage( DoubleTheDonationSettingsPage::class );
 
-		//Hooks::addFilter('plugin_action_links_' . GIVE_DTD_BASENAME, DoubleTheDonationSettingsPage::class, 'add_settings_link' );
+		Hooks::addFilter('plugin_action_links_' . GIVE_DTD_BASENAME, AddonServiceProvider::class, 'addSettingsLink' );
 
 		// Will display html of the import donation.
-		//Hooks::addAction('give_admin_field_dtd_intro',DoubleTheDonationSettingsPage::class, 'render_intro');
+		Hooks::addAction('give_admin_field_dtd_intro',AddonServiceProvider::class, 'renderIntro');
```

Comment:
@henryholtgeerts `Hooks::add*` delays class instantiation until the hook is called, so it isn't needed when hooking a method from within that class.

Reply:
Done, ref: https://github.com/impress-org/give-double-the-donation/pull/6/commits/91ebea18992d36c09ee6644bc91636de3f12ca09